core/textile/sharing.go

```diff
@@ -98,23 +98,29 @@ func (tc *textileClient) RejectSharedFilesInvitation(
 
 func (tc *textileClient) createReceivedFiles(
 	ctx context.Context,
-	invitationId string,
+	invitation domain.Invitation,
 	accepted bool,
-	paths []domain.FullPath,
-	keys [][]byte,
 ) error {
+	if len(invitation.ItemPaths) != len(invitation.Keys) {
```

Comment:
@perfectmak 

My comment would be that it might make sense to create a separate `struct` for sharing the keys and the items _together_, instead of matching them as arguments.

Then, instead of having a special type of empty `encryptionKey` it would be possible to mark the element as rejected.

Reply:
We could add the key as part of the `FullPath`, but that also means we have to find all the spots that's using this field and update it.  We could make a card for this and add it to upcoming sprints as well since it's working and doesn't affect functionality.

Reply:
Yeah, I was thinking the same too. It would take more work to refactor for now, so I'll create a card to track it.